runtime/src/accounts_db.rs

```diff
@@ -820,10 +915,44 @@ impl AccountsDB {
         }
     }
 
+    fn hash_accounts(&self, fork_id: Fork, accounts: &[(&Pubkey, &Account)]) -> Vec<Hash> {
+        let mut hashes = vec![];
+        let mut hash_state = BankHash::default();
+        let mut had_account = false;
+        for (pubkey, account) in accounts {
+            if !sysvar::check_id(&account.owner) {
+                let hash = BankHash::from_hash(&account.hash);
+                let new_hash = Self::hash_account(fork_id, account, pubkey);
+                let new_bank_hash = BankHash::from_hash(&new_hash);
+                debug!(
+                    "hash_accounts: key: {} xor {} current: {}",
+                    pubkey, hash, hash_state
+                );
+                if !had_account {
+                    hash_state = hash;
+                    had_account = true;
+                } else {
+                    Self::xor_hash_state(&mut hash_state, hash);
```

Comment:
could BankHash have an XOR method on it instead?

Reply:
Can do that.